ompi/mca/pml/ob1/pml_ob1_recvreq.h

```diff
@@ -196,9 +195,11 @@ recv_request_pml_complete(mca_pml_ob1_recv_request_t *recvreq)
             mca_bml_base_deregister_mem (recvreq->rdma_bml, recvreq->local_handle);
             recvreq->local_handle = NULL;
         }
+#if OPAL_ENABLE_MULTI_THREADS
+        opal_atomic_wmb();
+#endif
         MCA_PML_OB1_RECV_REQUEST_MPI_COMPLETE(recvreq);
     }
```

Comment:
Because this serialization was against a global mutex. All requests are now serialized based on their corresponding waiting context (via the sync_wait).